lib/migrations/migrate/Migrator.js

```diff
@@ -496,30 +502,40 @@ class Migrator {
       const migrationContent =
         this.config.migrationSource.getMigration(migration);
 
+      const beforeEach = this.config.beforeEach || (() => {});
+      const afterEach = this.config.afterEach || (() => {});
+
       // We're going to run each of the migrations in the current "up".
       current = current
         .then(async () => await migrationContent) //maybe promise
-        .then((migrationContent) => {
+        .then(async (migrationContent) => {
           this._activeMigration.fileName = name;
           if (
             !trx &&
             this._useTransaction(migrationContent, disableTransactions)
           ) {
             this.knex.enableProcessing();
-            return this._transaction(
-              this.knex,
-              migrationContent,
-              direction,
-              name
-            );
+            return await this.knex.transaction(async (trx) => {
+              await beforeEach(trx, [migration]);
```

Comment:
The migration that's passed into the hook is wrapped in an array here to maintain consistent types between the `each` and `all` hooks (instead of passing in an array in `all` but a migration object in `each`)